components/setting_item_max.tsx

```diff
@@ -1,131 +1,53 @@
     static defaultProps = {
         infoPosition: 'bottom',
         saving: false,
         section: '',
         containerStyle: '',
     };
+    private settingList: React.RefObject<HTMLDivElement>
 
-    static propTypes = {
-
-        /**
-         * Array of inputs selection
-         */
-        inputs: PropTypes.node,
-
-        /**
-         * Styles for main component
-         */
-        containerStyle: PropTypes.string,
-
-        /**
-         * Client error
-         */
-        clientError: PropTypes.oneOfType([
-            PropTypes.string,
-            PropTypes.object,
-        ]),
-
-        /**
-         * Server error
-         */
-        serverError: PropTypes.string,
-
-        /**
-         * Settings extra information
-         */
-        extraInfo: PropTypes.element,
-
-        /**
-         * Info position
-         */
-        infoPosition: PropTypes.string,
-
-        /**
-         * Settings or tab section
-         */
-        section: PropTypes.string,
-
-        /**
-         * Function to update section
-         */
-        updateSection: PropTypes.func,
-
-        /**
-         * setting to submit
-         */
-        setting: PropTypes.string,
-
-        /**
-         * Function to submit setting
-         */
-        submit: PropTypes.func,
-
-        /**
-         * Disable submit by enter key
-         */
-        disableEnterSubmit: PropTypes.bool,
-
-        /**
-         * Extra information on submit
-         */
-        submitExtra: PropTypes.node,
-
-        /**
-         * Indicates whether setting is on saving
-         */
-        saving: PropTypes.bool,
-
-        /**
-         * Settings title
-         */
-        title: PropTypes.node,
-
-        /**
-         * Settings width
-         */
-        width: PropTypes.string,
-
-        /**
-         * Text of cancel button
-         */
-        cancelButtonText: PropTypes.node,
-
-        /**
-         * Avoid submitting when using SHIFT + ENTER
-         */
-        shiftEnter: PropTypes.bool,
-
-        /**
-         * Text of save button
-         */
-        saveButtonText: PropTypes.string,
-    }
-
-    constructor(props) {
+    constructor(props: Props) {
         super(props);
         this.settingList = React.createRef();
     }
 
     componentDidMount() {
-        if (this.settingList.current) {
-            const focusableElements = this.settingList.current.querySelectorAll('.btn:not(.save-button):not(.btn-cancel), input.form-control, select, textarea, [tabindex]:not([tabindex="-1"])');
-            if (focusableElements.length > 0) {
-                focusableElements[0].focus();
-            } else {
-                this.settingList.current.focus();
-            }
+        const focusableElements = this.settingList?.current?.querySelectorAll?.('.btn:not(.save-button):not(.btn-cancel), input.form-control, select, textarea, [tabindex]:not([tabindex="-1"])');
```

Comment:
the first optional is not needed: 
```suggestion
        const focusableElements = this.settingList.current?.querySelectorAll?.('.btn:not(.save-button):not(.btn-cancel), input.form-control, select, textarea, [tabindex]:not([tabindex="-1"])');
```